gems/pending/VMwareWebService/VimTypes.rb

```diff
@@ -112,3 +124,22 @@ def initialize(vimObj)
     super(@vimFaultInfo.localizedMessage)
   end
 end
+
+class VimType
+  # Default to a sparse 1:1 mapping
+  @class_hierarchy = Hash.new { |hash, key| hash[key] = [key] }
+
+  # Add VmwareDistributedVirtualSwitch as a child class of DistributedVirtualSwitch
+  @class_hierarchy["DistributedVirtualSwitch"] << "VmwareDistributedVirtualSwitch"
+
+  def self.child_classes(vim_type)
+    @class_hierarchy[vim_type]
+  end
+
+  def self.base_class(vim_type)
+    @class_hierarchy.keys.each do |key|
+      return key if @class_hierarchy[key].include? vim_type
+    end
```

Comment:
This is expensive and will get worse as time goes because you are growing the @class_hierarchy on every call to child_classes with a new vim_type

I recommend instead to use `Set` instead of Array as the value type, so that lookups into that are fast.  Then, perhaps also add a reverse lookup table for @base_class, which you can use here.

```ruby

```